classes/migration/upgrade/v3_6_0/I857_ContributorRolesTypes.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_6_0/I857_ContributorRolesTypes.php
+ *
+ * Copyright (c) 2025 Simon Fraser University
+ * Copyright (c) 2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I857_ContributorRolesTypes
+ *
+ * @brief create the contributor roles table
+ */
+
+namespace PKP\migration\upgrade\v3_6_0;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use PKP\author\contributorRole\ContributorRoleIdentifier;
+use PKP\author\contributorRole\ContributorType;
+use PKP\facades\Repo;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\migration\Migration;
+use PKP\security\Role;
+use PKP\userGroup\UserGroup;
+
+class I857_ContributorRolesTypes extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        // Contributor roles
+        Schema::create('contributor_roles', function (Blueprint $table) {
+            $table->comment('The list of the contributor roles');
+            $table->bigInteger('contributor_role_id')->autoIncrement();
+            $table->bigInteger('context_id');
+            $table->enum('contributor_role_identifier', ContributorRoleIdentifier::getRoles());
+        });
+
+        // Contributor role settings
+        Schema::create('contributor_role_settings', function (Blueprint $table) {
+            $table->comment('Contributor role settings');
+            $table->bigInteger('contributor_role_setting_id')->autoIncrement();
+            $table->bigInteger('contributor_role_id');
+            $table->string('setting_name', 255);
+            $table->string('setting_value', 255);
+            $table->string('locale', 28);
+            $table->unique(['contributor_role_id', 'setting_name', 'locale'], 'contributor_role_id_setting_name_locale_unique');
+            $table->foreign('contributor_role_id', 'contributor_role_id_settings_foreign')->references('contributor_role_id')->on('contributor_roles')->onDelete('cascade');
+        });
+
+        // Add foreign key
+        Schema::table('credit_contributor_roles', function (Blueprint $table) {
+            $table->foreign('contributor_role_id', 'contributor_role_id_foreign')->references('contributor_role_id')->on('contributor_roles')->onDelete('cascade');
+        });
+
+        // Add author user group if not exist
```

Comment:
No need to do that -- I have just tested: the default user groups cannot be deleted.